packages/block-editor/src/components/typewriter/index.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * External dependencies
+ */
+import { debounce } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, createRef } from '@wordpress/element';
+import { computeCaretRect, getScrollContainer } from '@wordpress/dom';
+import { withSelect } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+
+class Typewriter extends Component {
```

Comment:
As a new component, I might have been curious to see if it could have been implemented as a function component using hooks (could simplify especially event binding with `useEffect` and handling with `useCallback`, cancellations on unmount, etc).

https://reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect

Reply:
I'm not sure how you'd convert the properties that I'm setting. E.g. `this.caretRect`. Using state here adds extra rerenders.

Reply:
Is this something blocking for this feature? I ask because it feels as a great UX improvement and it could always be refactored later in a separare issue/PR, right?

Reply:
I don't think this is blocking, but something nice to have. I'm just unsure how to do it properly. Might need some help there :D

Reply:
Properties are easily replaced with `useRef`

Reply:
How would that work?

Reply:
```
const  myProperty = useRef( 'initial value' );

// Using the value
console.log( myProperty.current );

// Assigning a new value
myProperty.current =  'some value';
```

Reply:
🤯 Didn't realise this could, or is designed to, be used for state as well.